src/parser.rs

```diff
@@ -851,6 +900,18 @@ impl Parser {
         Ok(values)
     }
 
+    /// Parse either `ALL` or `DISTINCT`. Returns `true` if `DISTINCT` is parsed and results in a
+    /// `ParserError` if both `ALL` and `DISTINCT` are fround.
+    pub fn parse_all_or_distinct(&mut self) -> Result<bool, ParserError> {
+        let all = self.parse_keyword("ALL");
+        let distinct = self.parse_keyword("DISTINCT");
+        if all && distinct {
+            return parser_err!("Cannot specify both ALL and DISTINCT".to_string());
```

Comment:
I mentioned this elsewhere, but this does reduce the info contained in the error messages by removing some context we previously had--I really don't know enough to say how important this is so please let me know what the preference is.

Reply:
As noted above, I think this is fine.